src/PolykeyClient.ts

```diff
@@ -15,20 +20,23 @@ import config from './config';
  * a new GRPCClientClient which attempts to connect to an existing PolykeyAgent's
  * grpc server.
  */
-interface PolykeyClient extends CreateDestroyStartStop {}
+// eslint-disable-next-line @typescript-eslint/no-unused-vars -- False positive for M
+interface PolykeyClient<M extends ClientManifest>
+  extends CreateDestroyStartStop {}
 @CreateDestroyStartStop(
   new errors.ErrorPolykeyClientRunning(),
   new errors.ErrorPolykeyClientDestroyed(),
 )
-class PolykeyClient {
-  static async createPolykeyClient({
+class PolykeyClient<M extends ClientManifest> {
+  static async createPolykeyClient<M extends ClientManifest>({
     nodeId,
     host,
     port,
     nodePath = config.defaults.nodePath,
     session,
-    grpcClient,
-    timer,
+    manifest,
+    connectionTimeout,
+    pingTimeout,
```

Comment:
How can these work? These are "transport" level properties, should not be part of `PolykeyClient`.

Reply:
This has been fixed.